### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by Ibnballo1**
In this pull request, we added unit tests to the following classes and their methods:

- Book
- Person
- Student
- Teacher
- Classroom
- Rental

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  end
+end
+
+# Test for author
+describe '#author' do
+  it 'should return the right author' do
+    book = Book.new('My Title', 'My Author')
+    expect(book.author).to eq('My Author')
+  end
+end
```
Comment (DammyShittu) [inline spec/book_spec.rb:25]: - [ ] Kindly import the Book class here as I was unable to run this test without importing it.

  ```
  require_relative '../book.rb'
  ```
  Or you can just import the `spec_helper.rb` file you defined.
  ```
  require_relative './spec_helper.rb'
  ```

  image

- [ ] All public methods should be tested. Kindly add tests for the `add_rental` method in the Book class.

File: spec/person_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  end
+end
+
+# Test for parent permission
+describe '#parent_permission' do
+  it 'should return the right parent permission' do
+    person = Person.new(16, 'My Name', 'Yes')
+    expect(person.parent_permission).to eq('Yes')
+  end
+end
```
Comment (DammyShittu) [inline spec/person_spec.rb]: - [ ] I was unable to run these tests as well. Kindly import the Person class or the `spec_helper.rb` file you defined.

- [ ] Kindly add tests for the `add_rental` and `can_use_services?` methods as contained in the Person class.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Ibnballo1 and @AlaaShaheen2 ,

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Munsa1 — APPROVED** [review]
Hi @Ibnballo1, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**